### ISSUE TITLE
Invalid Date for input date format YYYY-MM-DD h:i:s

### ISSUE THREAD

**Issue by scazzy** [felixge/node-dateformat#73]
I receive an Invalid date error for a properly valid date param mentioned in title. It's not UTC but it's a valid date

**Comment by atouzel** [comment]
Same here, this error if thrown only on Internet Explorer (11.1) : 

```
if (isNaN(date)) {
    throw TypeError('Invalid date');
}
```

and then IE can't even interpret it and throws this error in my console : 

> SCRIPT5022 You included a throw statement in your code, but it was not enclosed within a try block, or there was no associated catch block to trap the error.

**Comment by atouzel** [comment]
Don't know if that can fix your problem, but I fixed mine using the Data.parse() method in javascript : 

`dateFormat(Date.parse("2018-01-27 18:30:47"), "d mmmm yyyy");`

**Comment by yuconora** [comment]
In Safari/Webkit you can try YYYY/MM/DD instead YYYY-MM-DD

**Comment by Lelith** [comment]
another solution i found on stackoverflow was using moment

`var startDate = moment('2015-07-06 08:00', 'YYYY-MM-DD HH:mm').toDate();`

**Comment by patelakash317** [comment]
Try this: 
https://stackoverflow.com/a/50033029

**Comment by birendra2016** [comment]
> Don't know if that can fix your problem, but I fixed mine using the Data.parse() method in javascript :
> 
> `dateFormat(Date.parse("2018-01-27 18:30:47"), "d mmmm yyyy");`
This is wrong, what ever date will be passed, it will take always current date.

**Comment by pranabeshsingh** [comment]
Try this : 
var date = "2012-12-31".split("-");
var your_date = new Date(date[0], date[1]-1, date[2]);

**Comment by andrmoel** [comment]
> another solution i found on stackoverflow was using moment
> 
> `var startDate = moment('2015-07-06 08:00', 'YYYY-MM-DD HH:mm').toDate();`

Highly overloaded library. Works but I do not recommend to use moment when you want a light app.

**Comment by rmaximoc** [comment]
Someone is looking to this one?

**Comment by hamed-farag** [comment]
i got the same error too 
![image](https://user-images.githubusercontent.com/5382471/96868403-4914f380-146e-11eb-9c82-78dac7a95e3f.png)

and fixed by @alextouzel  solution


> 
> 
> Don't know if that can fix your problem, but I fixed mine using the Date.parse() method in javascript :
> 
> `dateFormat(Date.parse("2018-01-27 18:30:47"), "d mmmm yyyy");`

**Comment by osymo** [comment]
<img width="569" alt="Screen Shot 2022-02-26 at 19 42 05" src="https://user-images.githubusercontent.com/24529902/155857038-478bbd66-8bee-479f-ac6c-dcf0aec39699.png">

Using Date constructor on a string date with format **YYYY-MM-DD h:i:s**,  I got similar error with safari while the same code was working well with chrome.

The solution for me was to transform a format like this **YYYY-MM-DD h:i:s** to a format like this **YYYY-MM-DDTh:i:s** (notice the capital T between the date part and the time part).

I dit the transformation with a code like this: 

let d = '2021-09-12 00:00:00';

d = d.replace(' ', 'T')

and now when I do : new Date(d) everything works well.
